plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt

```diff
@@ -41,40 +45,87 @@ class PluginDeploymentListener(
 
         classes.forEach { (clazz, pluginAnnotation) ->
             try {
-                val pluginDefinition = deployPluginDefinition(
-                    PluginDefinition(pluginAnnotation.key, pluginAnnotation.title, pluginAnnotation.description, clazz.name)
-                )
+                val deployedPluginDefinition = createPluginDefinition(clazz, pluginAnnotation)
+
+                createActionDefinition(deployedPluginDefinition, clazz)
 
-                val actions = findPluginActions(clazz)
-                actions.forEach { (method, actionAnnotation) ->
-                    deployActionDefinition(
-                        PluginActionDefinition(
-                            PluginActionDefinitionId(
-                                actionAnnotation.key,
-                                pluginDefinition
-                            ),
-                            actionAnnotation.title,
-                            actionAnnotation.description,
-                            method.name,
-                            actionAnnotation.activityTypes.toList()
-                        )
-                    )
-                }
             } catch (e: Exception) {
                 throw PluginDefinitionNotDeployedException(pluginAnnotation.key, clazz.name, e)
             }
         }
     }
 
+    private fun createPluginDefinition(clazz: Class<*>, pluginAnnotation: Plugin): PluginDefinition {
+        val pluginProperties: MutableSet<PluginProperty> = mutableSetOf()
+        val pluginDefinition = PluginDefinition(
+            pluginAnnotation.key,
+            pluginAnnotation.title,
+            pluginAnnotation.description,
+            clazz.name,
+            pluginProperties
+        )
+
+        createProperties(pluginDefinition, pluginProperties, clazz)
+
+        return deployPluginDefinition(pluginDefinition)
+    }
+
+    private fun createProperties(
+        pluginDefinition: PluginDefinition,
+        pluginProperties: MutableSet<PluginProperty>,
+        clazz: Class<*>
+    ) {
+        val properties = findPluginProperties(clazz)
+        properties.forEach { (field, propertyAnnotation) ->
+            pluginProperties.add(
+                PluginProperty(
+                    PluginPropertyId(
+                        propertyAnnotation.key,
+                        pluginDefinition
```

Comment:
I'm wondering if we want to control this relation from one side. i.e. have the constructor call an pluginDefinition.addPluginProperty(this) method or something